pkg/restapi/device/operations.go

```diff
@@ -188,22 +188,98 @@ func (o *Operation) finishRegistration(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	credentialBytes, err := marshallResponse(credential)
+	device.AddCredential(*credential)
```

Comment:
For us, there's no sense in having an `AddCredential()` method on the device (see https://github.com/trustbloc/edge-agent/pull/417#discussion_r502611775)

Reply:
begin and finish login endpoints duo-lab library expects the credential, without that it doesnt work. I can discuss offline if it doesnt make sense

Reply:
@talwinder50 as per our discussion:

Agreed, let's leave the 1:N device:credential relationships until we understand the use cases better (such as the device registering multiple key types in different credentials).

TODO - understand these use cases better